celery_utils/models.py

```diff
@@ -60,3 +63,35 @@ def __str__(self):
             kwargs=self.kwargs,
             resolution='not resolved' if self.datetime_resolved is None else 'resolved'
         )
+
+
+class ChordData(models.Model):
+    """
+    A fairly simply class linking chords together.
+    """
+    completed_results = models.ManyToManyField(TaskMeta, related_name='chorddata_sub_results')
+    serialized_callback = models.TextField()  # A frozen, serialized callback signature
+    callback_result = models.OneToOneField(TaskMeta, related_name='chorddata_callback_result')
+    use_iterator = models.BooleanField(default=False)
+
+    def is_ready(self):
+        # we could do more here to detect/retry subtasks that failed
+        return all([result.status in READY_STATES for result in self.completed_results.all()])
+
+    def execute_callback(self):
+        callback_signature = Signature.from_dict(json.loads(self.serialized_callback))
+
+        if self.use_iterator:
+            # If we're using an iterator, it's assumed to be because there are size concerns with the results
+            # Thus, the callback_result TaskMeta object will have a null 'result' in the database, because you
+            # stored those results someplace else as part of the callback function, right?
```

Comment:
This block looks a bit strange - I pre-optimized a bit b/c I know "serialize all the results together in a list in memory" is a terrible idea for grade reports, where each result is a subset of a massive data file.